pyfr/backends/base/types.py

```diff
@@ -192,8 +207,11 @@ def __getitem__(self, idx):
     def __getattr__(self, attr):
         return getattr(self._curr_mat, attr)
 
-    def rslice(self, p, q):
-        raise RuntimeError('Matrix banks can not be sliced')
+    def slice(self, ra=None, rb=None, ca=None, cb=None):
```

Comment:
Could do 

class MatrixBank(Sequence):
    slice = MatrixBase.slice

above to avoid  duplicating slice method